### DISCUSSION TITLE
Instructions to deploy multi-node cluster w/o K8s

### DISCUSSION THREAD

**Discussion by ericsun2** [questdb/questdb#1221] [Ideas]
Wondering if there's a guide to deploy multiple OSS QuestDB containers via [Auto Scaling Groups](https://docs.aws.amazon.com/autoscaling/ec2/userguide/AutoScalingGroup.html) with Multi-Attach ESB volume to form a QuestDB cluster.

**Comment by bluestreak01** [comment]
hey Eric, we are testing this right now, i mean right now :) Trying to chose a file system that works. Please bear with us. Should have something for you very soon

**Reply by pde-rent** [reply]
Hi @bluestreak01 ! Any news on multi-node / horizontal scaling / HA ?
Also our RAID5 SSD storage infrastructure relies on a distributed SAN, which nodes are fiber wired and NFS connected, this seems to be a problem as QuestDB vectorization relies on the host mmap.
Do you foresee any way we could make QuestDB work in such a distributed environment ?
Many thanks

**Reply by bluestreak01** [reply]
We’re already making memory mapped files optional for writes. There is also plan to make mmap optional for reads too. We also started work on two-phase commit, which is precursor to HA. We will keep you posted here

**Reply by fnordian** [reply]
It's been a year. Any progress? ;-)

**Reply by ghost** [reply]
Hello, @bluestreak01 ,

Do you have any update about the questions above please?

Thanks in advance

**Reply by Javier-Godon** [reply]
Hello QuestDB team,

Thank you for all your work so far.

However, I must say that we are now in mid-2025, and it’s increasingly concerning that there is still no clear roadmap or commitment for replication or HA in the open source version.

From the outside, it feels like this critical feature is not a priority for the open source roadmap, and it would be helpful if you could be transparent about your plans.

Many of us are using QuestDB in production for serious workloads, and lack of clarity around replication and HA is becoming a blocker for continued adoption.

Could you please confirm:

Is replication or HA something that is realistically planned for the open source version in the foreseeable future?

Or should we assume that these features will only be available in an enterprise or closed-source offering, if at all?


I ask this respectfully, but directly, so that we can make informed architectural decisions for our critical systems rather than continue to wait indefinitely.

Thank you in advance for your honest clarification.

**Reply by bluestreak01** [reply]
Hi Javier, there are no plans right now to release HA as open source. This feature is available in the Enterprise edition. QuestDB Enterprise license comes at a cost. I am happy to put you in touch with our sales team if you wish.

**Reply by Javier-Godon** [reply]
Thank you for your quick and honest response.

I really appreciate the clarity. This helps us plan our architecture and align expectations with our operations and leadership teams.

**Comment by bluestreak01** [comment]
you are most welcome, please visit our website for more information https://questdb.com/enterprise/

**Reply by Javier-Godon** [reply]
The final decision on licensing isn’t mine, as I focus on the technical side, but I will share this information with our leadership and operations teams.

We really appreciate your transparency. We’ll review our options to see what best fits our requirements and architecture.

Thanks again for your support and for the great work you’re doing with QuestDB.

**Reply by bluestreak01** [reply]
Thank you for kind words Javier, I really appreciate that 🙏
